### PR TITLE
Dockerfile

### PR DESCRIPTION
**Pull request by janickoR**
- rewritten docker file in order to be more efficient
--> building docker image info:

stage | new time | old time | new mem | old mem |
--- | --- | --- | --- | --- |
build | **18.8s** | 45.9s | **~714 MB**  | ~ 1.82 GB |
dev | **16.6s** | 22.8s | **~501 MB** | ~ 1.88 GB |
prod | **20.2s** | 34.4s  | **~502 MB** | ~ 614 MB |

- package json and docker compose small changes
- added new endpoint for adding user to db - testing purpose (can be droped)

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -47,11 +49,9 @@
     "rimraf": "^5.0.1",
     "rxjs": "^7.5.6",
     "swagger-ui-express": "^5.0.0",
-    "update": "^0.7.4",
```
Comment (janickoR) [inline package.json:50]: this package contains critical vulnerabilities and i wasn't able to updgrade in satisfying way. Also i didn't find a case in which it is needed (if you know some pls tell me) so i decided to drop it.

File: Dockerfile [hunk 2]
```diff
@@ -1,60 +1,47 @@
...
-COPY package*.json ./
+ENTRYPOINT [ "/sbin/tini", "--" ]
 
-RUN npm install
-COPY . ./
+FROM base AS build-base
+WORKDIR /build
+COPY package.json package-lock.json ./
+COPY prisma ./prisma/
+RUN npm ci --omi=dev --frozen-lockfile
```
Comment (vidriduch) [inline Dockerfile:18]: Should this be just `--omi` and not `--omit` ?

File: Dockerfile [hunk 3]
```diff
@@ -1,60 +1,47 @@
...
 
+FROM build-base AS build
+COPY --chown=node:node . ./
 RUN npx prisma generate
+RUN npm run build
```
Comment (vidriduch) [inline Dockerfile:23]: Good question here actually. Should this really be two separate layers or just one? Meaning are these operations decoupled or they always need to be ran together?

More info on layers and what I mean: https://docs.docker.com/develop/develop-images/instructions/#run

**Review by vidriduch — CHANGES_REQUESTED** [review]
Given couple of comments that need to be discussed, but otherwise looks good

File: Dockerfile [hunk 4]
```diff
@@ -1,60 +1,46 @@
...
-COPY --chown=node:node --from=build /root/app/package*.json ./
-COPY --chown=node:node --from=build /root/app/node_modules ./node_modules
-RUN npm prune --production
-
-COPY --chown=node:node --from=build /root/app/dist ./dist
-COPY --chown=node:node --from=build /root/app/prisma ./prisma
+FROM app-base AS prod
+COPY --chown=node:node --from=build /build/package.json /build/package-lock.json ./
+COPY --chown=node:node --from=build /build/node_modules ./node_modules
+# RUN npm prune --production
```
Comment (vidriduch) [inline Dockerfile:36]: Why is this line here and commented out?

**Review by vidriduch — APPROVED** [review]
One more question but otherwise LGTM. Good job 👍

**Pull request merged**